### PR TITLE
Implement statistics tracking

### PR DESCRIPTION
**Pull request by CaptainElliott**

### CODE REVIEW

**Comment by CaptainElliott** [comment]
well fuck

File: src/main/java/in/twizmwaz/cardinal/module/modules/stats/Stats.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.entity.PlayerDeathEvent;
+
+import java.util.*;
+
+public class Stats implements Module {
+
+
+    private ArrayList<Tracker> stats = new ArrayList<>();
```
Comment (twizmwazin) [inline src/main/java/in/twizmwaz/cardinal/module/modules/stats/Stats.java]: It isn't a big deal, but generally one would make the field type List<> here, and then instantiate the ArrayList.

File: src/main/java/in/twizmwaz/cardinal/module/modules/stats/Tracker.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package in.twizmwaz.cardinal.module.modules.stats;
+
+import org.bukkit.OfflinePlayer;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+
+public class Tracker {
```
Comment (twizmwazin) [inline src/main/java/in/twizmwaz/cardinal/module/modules/stats/Tracker.java]: Spacing is messy in this class.

File: src/main/java/in/twizmwaz/cardinal/module/modules/stats/Stats.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+                    if (team != null && !team.isObserver()) {
+                        int teamKills = 0;
+                        int teamDeaths = 0;
+                        for (Tracker tracker : stats) {
+                            if (TeamUtils.getTeamByPlayer(tracker.getDamager()).equals(team)) teamKills++;
+                            if (TeamUtils.getTeamByPlayer(tracker.getDamaged()).equals(team)) teamDeaths++;
+                        }
+                        if (getPlayerWithMostKillsOnTeam(team) != null && getPlayerWithMostDeathsOnTeam(team) != null) {
+                            player.sendMessage("    " + team.getColor() + ChatColor.BOLD + team.getCompleteName() + ChatColor.GREEN + "    kills: " + ChatColor.WHITE + teamKills + ChatColor.RED + "    deaths     " + ChatColor.WHITE + teamDeaths + ChatColor.GOLD + " KD " + Math.round((double) teamKills / ((double) teamDeaths > 0 ? (double) teamDeaths : 1.0) * 100) / 100 );
+                            player.sendMessage("");
```
Comment (twizmwazin) [inline src/main/java/in/twizmwaz/cardinal/module/modules/stats/Stats.java]: I'm pretty sure sending "" does nothing.

File: src/main/java/in/twizmwaz/cardinal/module/modules/stats/Stats.java [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+public class Stats implements Module {
+
+
+    private ArrayList<Tracker> stats = new ArrayList<>();
+    private int totalKills;
+    private int totalDeaths;
+
+    @Override
+    public void unload() {
```
Comment (twizmwazin) [inline src/main/java/in/twizmwaz/cardinal/module/modules/stats/Stats.java]: HandlerList.unregisterAll(this);

File: src/main/java/in/twizmwaz/cardinal/module/modules/stats/Stats.java (hunk 3 above)
Comment (CaptainElliott) [inline src/main/java/in/twizmwaz/cardinal/module/modules/stats/Stats.java]: sends a blank line

Comment (twizmwazin) [inline src/main/java/in/twizmwaz/cardinal/module/modules/stats/Stats.java]: Does it actually? I'm pretty sure the Minecraft Client won't print a completely blank line.

Comment (CaptainElliott) [inline src/main/java/in/twizmwaz/cardinal/module/modules/stats/Stats.java]: http://i.imgur.com/b8Sbk4R.png

**Comment by AngelCasillas01** [comment]
@CaptainElliott add /stats if you can

**Comment by CaptainElliott** [comment]
It is on my todo list:)

**Pull request merged**